sklearn/utils/multiclass.py

```diff
@@ -545,8 +549,27 @@ def _ovr_decision_function(predictions, confidences, n_classes):
     k = 0
     for i in range(n_classes):
         for j in range(i + 1, n_classes):
-            sum_of_confidences[:, i] -= confidences[:, k]
-            sum_of_confidences[:, j] += confidences[:, k]
+            # Handle confidences as probabilities
+            # if probabilities are >= 0.5, (j is more probable) we should increase
+            # the sum of confidences of j and decrease the sum of confidences of i
+            if k in probabilities_indexes:
```

Comment:
I assume that if we use `_get_response_values_binary`, we can be aware of the method use so it should be easier to handle what happens here.

Reply:
Thank you! I will take a look at it, but this will only be possible after the refactor, correct?